site/components/src/main/java/uk/nhs/digital/intranet/components/SearchPageComponent.java

```diff
@@ -21,17 +47,90 @@ public SearchPageComponent(final GraphProvider graphProvider) {
     @Override
     public void doBeforeRender(HstRequest request, HstResponse response) {
         super.doBeforeRender(request, response);
-        final String searchQuery = getSearchQuery(request);
-        final List<SearchResult> results = graphProvider.getPeople(searchQuery);
-        populateRequest(request, results);
+        final List<SearchResult> searchResults = new ArrayList<>();
+        searchResults.addAll(getBloomreachResults(request));
+        searchResults.addAll(graphProvider.getPeople(getSearchQuery(request)));
+
+        //TODO sorting searchResults list
+
+        Pageable<SearchResult> pageable = getPageableFactory()
+            .createPageable(searchResults, getCurrentPage(request),
+                getComponentInfo(request).getPageSize());
+        populateRequest(request, pageable);
     }
 
-    private void populateRequest(final HstRequest request, final List<? extends SearchResult> results) {
+    private void populateRequest(final HstRequest request,
+        final Pageable<SearchResult> pageable) {
         request.setAttribute(REQUEST_ATTR_QUERY, getSearchQuery(request));
-        request.setModel(REQUEST_ATTR_RESULTS, results);
+        request.setAttribute(REQUEST_ATTR_PAGEABLE, pageable);
     }
 
     protected String getSearchQuery(HstRequest request) {
         return cleanupSearchQuery(getAnyParameter(request, REQUEST_PARAM_QUERY));
     }
+
+    protected int getCurrentPage(HstRequest request) {
+        return getAnyIntParameter(request, REQUEST_PARAM_PAGE, 1);
+    }
+
+    protected EssentialsListComponentInfo getComponentInfo(HstRequest request) {
+        return getComponentParametersInfo(request);
+    }
+
+    private List<SearchResult> getBloomreachResults(HstRequest request) {
```

Comment:
Can you extract this into its own class and make it a bean?

Reply:
I think it's going to be better if we separate them, and potentially use some `CompleteableFuture`s to run the 2 search queries at the same time

Reply:
that makes sense